x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/Ccr.java

```diff
@@ -272,25 +276,24 @@ public List<RestHandler> getRestHandlers(
             return emptyList();
         }
 
-        return Arrays.asList(
-            // stats API
-            new RestFollowStatsAction(),
-            new RestCcrStatsAction(),
-            new RestFollowInfoAction(),
-            // follow APIs
-            new RestPutFollowAction(),
-            new RestResumeFollowAction(),
-            new RestPauseFollowAction(),
-            new RestUnfollowAction(),
-            // auto-follow APIs
-            new RestDeleteAutoFollowPatternAction(),
-            new RestPutAutoFollowPatternAction(),
-            new RestGetAutoFollowPatternAction(),
-            new RestPauseAutoFollowPatternAction(),
-            new RestResumeAutoFollowPatternAction(),
-            // forget follower API
-            new RestForgetFollowerAction()
-        );
+        return Stream.concat(
```

Comment:
`Arrays.asList` returns a fixed-size list. Ass a result, adding later the additional handler based on the value of `Build.current().isSnapshot()` might fail with an `UnsupportedOperationException`.

Reply:
Maybe just create an array list with an if statement for the shard changes action? Looks simpler to me than all this stream stuff.